backend/api/database.py

```diff
@@ -50,3 +50,34 @@ def delete_db_user(db: Session, id: int):
     rows_deleted = db.query(UserAccount).filter(UserAccount.id == id).delete()
     db.commit()
     return rows_deleted
+
+
+def get_db_districts(db: Session, user):
+    if user["is_admin"]:
```

Comment:
This is sort of an implementation decision, but I would actually not leak the notion of user roles to the database layer. Instead, it might be "cleaner" to have this helper function support an optional argument like `active_only=True` which then gets passed or not at the request handler so the business logic is baked in there. This, for example, would allow use of this function from any context in the future where the notion of `user` may not even exist.

Reply:
Resolved in latest commit.

Reply:
Check my :eyes: , but I think the logic is inverted in both `main.py` and `database.py` (e.g. it works as expected, but it reads wrong). I think in `main.py` it should be:

```python
    active_only = not current_user["is_admin"]
```

So if you're not an admin, you get only active districts. Similarly, in `database.py`:

```python
    if not active_only:
        school_districts = db.query(SchoolDistrict).all()
```

Reply:
Ahhhh, I see now. Fixed.